src/components/icons/Icon/Icon.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
+import styles from './Icon.module.scss'
+
+export type IconProps = React.SVGAttributes<SVGElement> & {
+    className?: string;
+    color?: 'primary' | 'secondary' | 'accent'; 
+width?: number;
+    height?: number;
+};
+
+const Icon: React.FC<React.PropsWithChildren<IconProps>> = ({color,
+    width = 24,
+    height = 24,
+    className,
+    children,
+    ...props}) => {
+      const colorClass = styles[`icon_${color}`];
+        return(
+            <svg
+            width={width}
+            height={height}
+           
+            className={`${colorClass} ${className || ''} `}
```

Comment:
[+] в таких случаях нужно использовать либу classnames
```suggestion
            className={classNames(colorClass, className)}
```